apps/etterlatte-tidshendelser/src/main/kotlin/no/nav/etterlatte/tidshendelser/HendelseDao.kt

```diff
@@ -0,0 +1,196 @@
+package no.nav.etterlatte.tidshendelser
+
+import kotliquery.Row
+import kotliquery.TransactionalSession
+import kotliquery.queryOf
+import no.nav.etterlatte.libs.common.tidspunkt.toLocalDatetimeUTC
+import no.nav.etterlatte.libs.database.Transactions
+import no.nav.etterlatte.libs.database.tidspunkt
+import no.nav.etterlatte.libs.database.transaction
+import org.slf4j.LoggerFactory
+import java.time.YearMonth
+import javax.sql.DataSource
+
+class HendelseDao(private val datasource: DataSource) : Transactions<HendelseDao> {
+    private val logger = LoggerFactory.getLogger(HendelseDao::class.java)
+
+    override fun <R> inTransaction(block: HendelseDao.(TransactionalSession) -> R): R =
+        datasource.transaction {
+            this.block(it)
+        }
+
+    fun hentJobb(id: Int): HendelserJobb {
+        return datasource.transaction { tx ->
+            queryOf("SELECT * FROM jobb WHERE id = :id", mapOf("id" to id))
+                .let { query -> tx.run(query.map { row -> row.toHendelserJobb() }.asSingle) }
+                ?: throw NoSuchElementException("Fant ikke jobb med id $id")
+        }
+    }
+
+    fun finnAktuellJobb(): List<HendelserJobb> {
+        return datasource.transaction { tx ->
+            queryOf(
+                """
+                SELECT * FROM jobb 
+                WHERE status = :status
+                AND kjoeredato = CURRENT_DATE
+                ORDER BY id asc
+                LIMIT 1
+                """.trimIndent(),
+                mapOf("status" to "NY"),
+            )
+                .let { query -> tx.run(query.map { row -> row.toHendelserJobb() }.asList) }
+        }
+    }
+
+    fun hentJobber(jobbIDs: List<Int>): List<HendelserJobb> {
+        return datasource.transaction { tx ->
+            queryOf(
+                """
+                SELECT * FROM jobb WHERE id = ANY(?)
+                """.trimIndent(),
+                jobbIDs.toTypedArray(),
+            )
+                .let { query -> tx.run(query.map { row -> row.toHendelserJobb() }.asList) }
+        }
+    }
+
+    fun oppdaterJobbstatusStartet(hendelserJobb: HendelserJobb) {
+        oppdaterJobbstatus(hendelserJobb, "STARTET")
```

Comment:
Kanskje lurt med enum/konstantar for jobbstatus også.

Reply:
døn